lib/spack/spack/cmd/install.py

```diff
@@ -294,8 +305,17 @@ def install(parser, args, **kwargs):
                     # once, as it can be slow.
                     env.write(regenerate_views=False)
 
-            tty.msg("Installing environment %s" % env.name)
-            env.install_all(args)
+            specs = env.all_specs()
+            if not args.log_file and not reporter.filename:
+                reporter.filename = default_log_file(specs[0])
```

Comment:
@tldahlgren environments don't always have a nonzero number of specs, and then `spack install` fails. Could you potentially fix this? I'm not sure what `reporter` is.

Reply:
There's a quick change (or fix?) at #28031